server/src/main/java/org/elasticsearch/cluster/metadata/AliasOrIndex.java

```diff
@@ -153,6 +175,24 @@ public void computeAndValidateWriteIndex() {
                 throw new IllegalStateException("alias [" + aliasName + "] has more than one write index [" +
                     Strings.collectionToCommaDelimitedString(writeIndicesStrings) + "]");
             }
+
+            // Validate hidden status
+            final Map<Boolean, List<IndexMetaData>> groupedByHiddenStatus = referenceIndexMetaDatas.stream()
```

Comment:
If I understand this correctly, this means that every place the alias is stored must have the same hidden value? Should we be more strict regarding false and null since it seems like a combination of those values would be accepted?

Reply:
Yes, every `IndexMetaData` that has an alias with a particular name must have the same logical value for `is_hidden`.  I thought about making it more strict about false vs null, but given that the default for this value is `false`, it would seem odd that adding an alias that is explicitly not hidden (`is_hidden: false`), when the alias already exists as non-hidden by default (`is_hidden` omitted or null), would fail.

I would certainly be open to being more strict about false vs null, but I'm not sure what it would gain us, and it's a little unintuitive to be strict about explicitly false vs omitted with a false default.

Reply:
I haven't read all the code in detail but just wanted to make sure that this check won't prevent an existing alias that spans multiple indices from being made hidden.

Suppose we have this situation in 7.6:

* `.ml-annotations-1` and `.ml-annotations-2` are indices
* `.ml-annotations-read` is an alias on both `.ml-annotations-1` and `.ml-annotations-2`
* `.ml-annotations-write` is an alias on just `.ml-annotations-2`

In 7.7 we want to make all this hidden, so we:

* Add the `index.hidden` setting to `.ml-annotations-1` and `.ml-annotations-2`
* Add the `is_hidden` property to `.ml-annotations-read` and `.ml-annotations-write`

I just wanted to check that this won't fall foul of this validation because there's an instant during the updates where `.ml-annotations-read` is only hidden on `.ml-annotations-1` but not `.ml-annotations-2`, even though it would be applied consistently an instant later.

Reply:
@droberts195 What you'll have to do is update the alias on all indices that have that alias simultaneously, like so:
```
POST _aliases
{
    "actions": [
        {
            "add": {
                "index": ".ml-annotations-1",
                "alias": ".ml-annotations-read",
                "is_hidden": true
            }
        },
        {
            "add": {
                "index": ".ml-annotations-2",
                "alias": ".ml-annotations-read",
                "is_hidden": true
            }
        }
    ]
}
```

Even though the action is `add`, you can use it to update aliases that already exist. This is a bit inconvenient, but it's somewhat of a limitation of the alias infrastructure, and we should be able to work around it by querying for the list of indices and alias points to and generating the update aliases list from that. Let me know if this would be troublesome for you/your team.

Reply:
Thanks @gwbrown.  I think doing the update as you've shown should be OK for us.